### PR TITLE
ISLE: Resolve overlap in the riscv64 backend

### PR DESCRIPTION
**Pull request by elliottt**
Resolve overlap in the RiscV64 backend by adding priorities to rules. Additionally, one test updated as a result of this work, as a peephole optimization for addition with immediates fires now.

### CODE REVIEW

File: cranelift/codegen/src/isa/riscv64/inst.isle [hunk 1]
```diff
@@ -911,17 +911,17 @@
   (if-let $false(has_b))
   (gen_cltz $false x ty))
 
-(rule
+(rule 2
   (lower_ctz $I64 x)
   (if-let $true(has_b))
```
Comment (cfallin) [inline cranelift/codegen/src/isa/riscv64/inst.isle:916]: pre-existing but can we fix the formatting here while we're here (space after `$true`) and twice below?

File: cranelift/codegen/src/isa/riscv64/inst.isle [hunk 2]
```diff
@@ -1011,12 +1011,12 @@
 
 ;; val is_signed from_bits to_bits
 (decl lower_extend(Reg bool u8 u8)ValueRegs)
```
Comment (cfallin) [inline cranelift/codegen/src/isa/riscv64/inst.isle:1013]: likewise here -- pre-existing but would you mind doing a search for `[A-Za-z0-9](` and `)[A-Za-z0-9]` and fixing spaces around paren-forms?

(happy to do this in a separate PR if you'd rather get this in as-is but if it's easy...)

**Review by cfallin — APPROVED** [review]
Looks great overall, thanks!

A few pre-existing formatting nits I noticed while reading this but we can handle them separately if you prefer.

**Comment by elliottt** [comment]
Happy to fix the formatting nits in this PR. I held off originally to make sure the diff was minimal :)
